packages/angular_devkit/build_angular/src/angular-cli-files/plugins/single-test-transform.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { logging } from '@angular-devkit/core';
+import { getOptions } from 'loader-utils';
+import { loader } from 'webpack';
+
+export interface SingleTestTransformLoaderOptions {
+  files: string[];
+  logger: logging.Logger;
+}
+
+export default function loader(this: loader.LoaderContext, source: string) {
+  const options = getOptions(this) as SingleTestTransformLoaderOptions;
+  const start = 'import \'';
+  const end = '\';';
+  const testCode = start + options.files
+    .map(path => `./${path.replace('.ts', '')}`)
+    .join(`${end}\n${start}`) + end;
+
+  // TODO: maybe a documented 'marker/comment' inside test.ts would be nicer?
```

Comment:
For our use case at work it would be also nice to be able to remove some imports / matching regexes when running specific test. We need to exclude an import that includes angularjs and basically entire project for old unit tests to pass. Should I cover such use case here? 

I am fine with monkey patching this script in post install until we finish upgrading, so personally it would not block me/us from using this feature.

Reply:
I think that would be a bit too much right now... but an easy alternative is to have an alternate test configuration in `angular.json` that uses a different `main` for tests.